### ISSUE TITLE
Advance next page does not work when I switch between amount of data to display

### ISSUE THREAD

**Issue by AngelCareaga** [bhaveshpatel200/ng-datatable#3]
There seems to be a small bug when switching between the amount of data to display, as after doing so and clicking next it doesn't seem to respond until it is **clicked twice**.

[Grabación de pantalla desde 2024-05-18 16-41-27.webm](https://github.com/bhaveshpatel200/ng-datatable/assets/8092661/4569ed51-bc26-4952-9f19-88ab80211d23)

**Comment by bhaveshpatel200** [comment]
Hi,

Its not a bug, buttons will be disabled till response not completed. you clicked before response completed, so you feel like its not working on first click but actually its works.

**Comment by AngelCareaga** [comment]
@bhaveshpatel200 

It does not seem to be that, you can verify in the implementation of the following link:

https://angular.vristo.sbthemes.com/datatables/basic

In which when you change from 10 to 20 and click the next button (not the number), it does not work until you click it twice.

I already tried waiting a while without clicking, about 5 seconds and it works. The curious thing is that if we click on any number, even immediately it works, it is strange.

Translated with DeepL.com (free version)

**Comment by bhaveshpatel200** [comment]
Yes got it, will check once getting time and update in next release.

**Comment by bhaveshpatel200** [comment]
Can you please check its working in our demo.
https://ng-datatable-document.vercel.app/static-mode
